src/hyperResource.js

```diff
@@ -94,14 +122,15 @@ function isTemplated( url ) { // jshint ignore:line
 
 function getActionUrlCache( resources, prefix, version ) {
 	var cache = {};
-	var parentUrlFn = getParentUrlFn( resources, prefix, version );
+	var parentUrlFn = getParentUrlFn( resources, version );
+	var prefixFn = getPrefix.bind( undefined, resources, prefix );
 
 	_.reduce( resources, function( rAcc, resource, resourceName ) {
 		resource = getVersion( resource, version );
 		rAcc[ resourceName ] = _.reduce( resource.actions, function( acc, action, actionName ) {
 			var actionSegment = resource.actions[ actionName ].url;
 			var resourceSegment = getResourcePrefix( actionSegment, resource, resourceName );
-			var actionUrl = [ resource.urlPrefix, resourceSegment, actionSegment ].join( "" ).replace( "//", "/" );
+			var actionUrl = [ resourceSegment, actionSegment ].join( "" ).replace( "//", "/" );
```

Comment:
Just a suggestion, but Node has a [`url` module](https://nodejs.org/api/url.html) that would handle this for you.

```js
var url = require( "url" );
var actionUrl = url.resolve( resourceSegment, actionSegment );
```
Please ignore If you already knew that and there was a reason it didn't work here.